libsolidity/formal/CHC.cpp

```diff
@@ -681,7 +673,9 @@ void CHC::resetSourceAnalysis()
 	m_errorIds.clear();
 	m_callGraph.clear();
 	m_summaries.clear();
-	m_symbolFunction.clear();
+	m_interfaces.clear();
+	m_nondetInterfaces.clear();
+	Predicate::reset();
```

Comment:
By the way: I'd say it's fine for now, but in general, it'd be nicer to have explicit ownership here (i.e. make the map a member of ``CHC`` like all the others above and just clear it here, making ``Predicate::create`` a member of ``CHC`` instead... in which case a public constructor in ``Predicate``, but it being non-movable makes more sense...)... but I don't think it's worth changing this now (and you probably had reasons not to go that way anyways).

Reply:
The reason to not go that way is that I kinda wanted to decouple CHC from Predicates. But also wouldn't mind it the way you described.

Reply:
The thing is: static members with resets like this are always a bit hacky IMO... and at the latest if we ever want the compiler to be re-entrant (i.e. we allow multiple instances of the compiler to live in a process) all of them have to go... maybe we'll never do that, though :-)... but yeah, fine to keep it for now.